### PR TITLE
UIOR-1195: Add custom fields to views

### PR DESCRIPTION
**Pull request by elsenhans**
## Purpose
This JIRA ticket is about integrating the current folio-custom-fields functionality into the ui-orders application. This integration will involve implementing the functionality on both the Purchase Order (PO) and Purchase Order Line (POL) levels. The work has been divided into multiple tickets to make it more manageable and organized.

**Requirements:**
- Add Custom Fields components to View/Edit/Create panes of Purchase Orders
-  Add Custom Fields components to View/Edit/Create panes of Purchase Order Lines
- Add Custom Fields components to View/Edit/Create panes of Purchase Order Templates
- Create/Update tests

https://issues.folio.org/browse/UIOR-1195

### CODE REVIEW

**Comment by alisher-epam** [comment]
@NikitaSedyx Is the PR base branch should be `develop`?

**Comment by NikitaSedyx** [comment]
@alisher-epam  no

**Comment by NikitaSedyx** [comment]
@folio-org/fe-tl-reviewers please take a look at this one, thank you

**Review by ryandberger — APPROVED** [review]

**Comment by elsenhans** [comment]
@NikitaSedyx do I need to wait for another approval from a second reviewer?

**Comment by NikitaSedyx** [comment]
@elsenhans yes, please, I pinged team to review

File: src/components/POLine/POLineForm.js
Comment (alisher-epam) [inline src/components/POLine/POLineForm.js:17]: Can you order the exports in alphabetical order, please?
```suggestion
import { Field } from 'react-final-form';
import { FormattedMessage } from 'react-intl';
import { useHistory } from 'react-router';
```

File: src/components/POLine/POLineForm.js
Comment (alisher-epam) [inline src/components/POLine/POLineForm.js:70]: the same ordering

File: src/components/POLine/POLineForm.js
Comment (alisher-epam) [inline src/components/POLine/POLineForm.js:616]: Please declare a variable above. Ex: https://github.com/folio-org/ui-orders/blob/4c58a9cf9843ebd534813dd3ef7e90d4ce6cf36f/src/components/POLine/POLineForm.js#L396

File: src/components/POLine/POLineForm.test.js
Comment (alisher-epam) [inline src/components/POLine/POLineForm.test.js:164]: Regex is redundant here. "EditCustomFieldsRecord" will print the same result

File: src/components/POLine/POLineView.test.js
Comment (alisher-epam) [inline src/components/POLine/POLineView.test.js:122]:
```suggestion
    expect(screen.getByText('ViewCustomFieldsRecord')).toBeInTheDocument();
```

File: src/components/PurchaseOrder/POForm.js
Comment (alisher-epam) [inline src/components/PurchaseOrder/POForm.js:7]: import order please

File: src/components/PurchaseOrder/POForm.js
Comment (alisher-epam) [inline src/components/PurchaseOrder/POForm.js:40]: order

File: src/components/PurchaseOrder/POForm.js
Comment (alisher-epam) [inline src/components/PurchaseOrder/POForm.js:385]: variable

File: src/components/PurchaseOrder/POForm.test.js
Comment (alisher-epam) [inline src/components/PurchaseOrder/POForm.test.js:115]: ^^

File: src/settings/CustomFieldsSettings.js
Comment (alisher-epam) [inline src/settings/CustomFieldsSettings.js:8]: Leave a empty line

File: src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js
Comment (alisher-epam) [inline src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js:52]: order

**Comment by usavkov-epam** [comment]
Hi, @elsenhans, great work! I have a question about the accordion labels for the custom fields. It seems that if I haven't touched the label in the settings, it displays 'Custom Fields'. However, for example, in the forms, it's empty. If I change the label in the settings, it starts to work. Could there be something missing?


https://github.com/folio-org/ui-orders/assets/88109087/9f9d183b-9394-4d25-80fb-a57be4440583

File: src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js
Comment (alisher-epam) [inline src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js:24]: fix the order, please. General rule: Import orders should be in hierarchical and alphabetic
Ex: 
```
import constants from '../../constants';
import anotherConstants from '../constants'
import somePackage from '../utils'
```
external packages
--- leave an empty line ---
folio packages
--- leave an empty line ---
internal packages

File: src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js
Comment (alisher-epam) [inline src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js:419]: varibale

File: src/components/POLine/POLineForm.js
Comment (usavkov-epam) [inline src/components/POLine/POLineForm.js:97]: "Game changer" fields are basically fields, which influence other fields in the form. Do the `customFields` field is one of them?

Comment (elsenhans) [reply src/components/POLine/POLineForm.js:97]: @usavkov-epam  Thanks for your response!
I need to add `customFields` to `GAME_CHANGER_FIELDS` to transfer the customFields values defined in the template.
Is there another way to do that?

Comment (usavkov-epam) [reply src/components/POLine/POLineForm.js:97]: At first glance, it looks like the values should be taken from the template without changing the game-changer array. For all registered form fields (when the change game fields played a role), a check is carried out for the presence of a value in the associated template; if there is one, it is substituted into the form.

ref https://github.com/folio-org/ui-orders/blob/master/src/components/POLine/POLineForm.js#L199

If you leave the `GAME_CHANGER_FIELDS` untouched, do you have problems with the population of custom fields from the template?

Comment (elsenhans) [reply src/components/POLine/POLineForm.js:97]: Exactly, if i leave `GAME_CHANGER_FIELDS` untouched, the `customFields` values from the templates are not taken over to the form.

Comment (usavkov-epam) [reply src/components/POLine/POLineForm.js:97]: Hmm, strange, it works for me locally.
But if your approach works it's acceptable I think

Comment (elsenhans) [reply src/components/POLine/POLineForm.js:97]: Well, it works for `POForm` but not for `POLineForm`

**Comment by elsenhans** [comment]
> Hi, @elsenhans, great work! I have a question about the accordion labels for the custom fields. It seems that if I haven't touched the label in the settings, it displays 'Custom Fields'. However, for example, in the forms, it's empty. If I change the label in the settings, it starts to work. Could there be something missing?
> chrome_NEqjbZaF2p.mp4

Seems like the default title for the Accordion in `EditCustomFieldsRecord` is missing. 
I created a Ticket for this bug:
https://issues.folio.org/browse/STSMACOM-805

File: src/components/POLine/POLineForm.js
Comment (alb3rtino) [reply src/components/POLine/POLineForm.js:97]: The problem here is the asynchronous nature of `<EditCustomFieldsRecords>`, which renders its `<Field>` components only after successfully fetching data from the custom-fields API. The `populateFieldsFromTemplate()` method may be invoked before these `<Field>` components have registered with the form, leading to inconsistent behavior. Even if `"customFields"` is added to `GAME_CHANGER_FIELDS`, it does not guarantee that the fields are present when attempting to populate them, resulting in a race condition.

Comment (elsenhans) [reply src/components/POLine/POLineForm.js]: Thanks @alb3rtino for the hint! 
`customFields` is no longer a `GAME_CHANGER_FIELDS`, I removed it.
We have to add a callback function to `<EditCustomFieldsRecords>` so we know when the values are loaded.

**Review by usavkov-epam — APPROVED** [review]

**Comment by elsenhans** [comment]
@alisher-epam Are the adjustments okay for you as well?

File: src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js
Comment (alisher-epam) [inline src/settings/OrderTemplates/OrderTemplateView/OrderTemplateView.js:28]: You can remove it. The whole idea is a create a logical group for imports.
// Extrenal

// Folio

// Internal

File: src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js
Comment (alisher-epam) [inline src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js:25]: Empty line here

**Review by alisher-epam — APPROVED** [review]
Except empty line feedback, the rest looks good.

File: src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js
Comment (elsenhans) [reply src/settings/OrderTemplates/OrderTemplatesEditor/OrderTemplatesEditor.js:25]: Are you sure? 
Here is the grouping like:
import ... from 'prop-types';
import ... from 'prop-types';
import ... from 'prop-types';

**Pull request merged**
